app/services/recent-events.ts

```diff
@@ -0,0 +1,273 @@
+    }
+    return $t('has subscribed (%{tier})', { tier: subscriptionMap(event.sub_plan) });
+  }
+
+  onSocketEvent(e: TSocketEvent) {
+    if (SUPPORTED_EVENTS.includes(e.type)) {
+      this.onEventSocket(e as IEventSocketEvent);
+    }
+  }
+
+  onEventSocket(e: IEventSocketEvent) {
+    e.message.forEach((msg: IRecentEvent) => (msg.type = e.type));
+    this.ADD_RECENT_EVENT(e.message);
+  }
+
+  getEventString(event: IRecentEvent) {
+    return {
+      donation:
+        $t('has donated') +
+        (event.crate_item ? $t(' with %{name}', { name: event.crate_item.name }) : ''),
+      merch: $t('has purchased %{product} from the store', { product: event.product }),
+      follow: $t('has followed'),
+      subscription: this.getSubString(event),
+      // Twitch
+      bits: $t('has used'),
+      host: $t('has hosted you with %{viewers} viewers', { viewers: event.viewers }),
+      raid: $t('has raided you with a party of %{viewers}', { viewers: event.raiders }),
+      // Mixer
+      sticker: $t('has used %{skill} for', { skill: event.skill }),
+      effect: $t('has used %{skill} for', { skill: event.skill }),
+      // Facebook
+      like: $t('has liked'),
+      stars: $t('has used'),
+      support: $t('has supported for %{mounths} months', { months: event.months }),
+      share: $t('has shared'),
+      // Youtube
+      superchat: $t('has superchatted'),
+      // Integrations
+      pledge: $t('has pledged on Patreon'),
+      eldonation: $t('has donated to ExtraLife'),
+      tiltifydonation: $t('has donated to Tiltify'),
+      donordrivedonation: $t('has donated to Donor Drive'),
+      justgivingdonation: $t('has donated to Just Giving'),
+      treat: $t('has given a treat %{title}', { title: event.title }),
+    }[event.type];
+  }
+
+  async toggleMuteEvents() {
+    const headers = authorizedHeaders(
+      this.userService.apiToken,
+      new Headers({ 'Content-Type': 'application/json' }),
+    );
+    const url = `https://${
+      this.hostsService.streamlabs
+    }/api/v5/slobs/widget/recentevents/eventspanel`;
+    const body = JSON.stringify({ muted: !this.state.muted });
+    return await fetch(new Request(url, { headers, body, method: 'POST' }))
+      .then(handleResponse)
+      .then(() => this.SET_MUTED(!this.state.muted));
+  }
+
+  openRecentEventsWindow(isMediaShare?: boolean) {
+    this.windowsService.createOneOffWindow(
+      {
+        componentName: 'RecentEvents',
+        queryParams: { isMediaShare },
+        title: $t('Recent Events'),
+        size: { width: 800, height: 600 },
+      },
+      'RecentEvents',
+    );
+  }
+
+  @mutation()
+  ADD_RECENT_EVENT(events: IRecentEvent[]) {
+    this.state.recentEvents = events.concat(this.state.recentEvents);
+  }
+
+  @mutation()
+  SET_RECENT_EVENTS(eventArray: IRecentEvent[]) {
+    this.state.recentEvents = eventArray;
+  }
+
+  @mutation()
+  SET_MUTED(muted: boolean) {
+    this.state.muted = muted;
+  }
+}
```

Comment:
Please mark as `private` the all methods that are not calling outside this file. It provides better IntelliSense and refactoring capabilities